File: Java/src/RomanToInteger.java
import java.util.HashMap;
import java.util.Map;

/**
 * Given a roman numeral, convert it to an integer.
 * Input is guaranteed to be within the range from 1 to 3999.
 * <p>
 * Accepted.
 */
public class RomanToInteger {

    public int romanToInt(String s) {
        if (s == null) {
            return -1;
        }

        Map<Character, Integer> dict = new HashMap<>(7);
        dict.put('I', 1);
        dict.put('V', 5);
        dict.put('X', 10);
        dict.put('L', 50);
        dict.put('C', 100);
        dict.put('D', 500);
        dict.put('M', 1000);

        char[] chars = s.toCharArray();
        int result = 0;

        for (int i = 0; i < chars.length; i++) {
            if (i + 1 < chars.length
                    && dict.get(chars[i + 1]) > dict.get(chars[i])
                    && (chars[i] == 'I' || chars[i] == 'X' || chars[i] == 'C')) {
                result += dict.get(chars[i + 1]) - dict.get(chars[i]);

                i++;
            } else {
                result += dict.get(chars[i]);
            }
        }

        return result;
    }

}

File: Kotlin/src/RomanToInteger.kt
/**
 * Given a roman numeral, convert it to an integer.
 * Input is guaranteed to be within the range from 1 to 3999.
 *
 * Accepted.
 */
class RomanToInteger {

    fun romanToInt(s: String): Int {
        val dict = mapOf(
                Pair('I', 1),
                Pair('V', 5),
                Pair('X', 10),
                Pair('L', 50),
                Pair('C', 100),
                Pair('D', 500),
                Pair('M', 1000))

        val chars = s.toCharArray()
        var result = 0
        var i = 0
        while (i < chars.size) {
            if (i + 1 < chars.size
                    && (dict[chars[i + 1]]!! > dict[chars[i]]!!)
                    && (chars[i] == 'I' || chars[i] == 'X' || chars[i] == 'C')) {
                result += dict[chars[i + 1]]!! - dict[chars[i]]!!

                i++
            } else {
                result += dict[chars[i]]!!
            }
            i++
        }

        return result
    }

}